[new product] Deno (#6449)
captn3m0/endoflife.date: products/deno.md

---
title: Deno
category: framework
tags: javascript-runtime
permalink: /deno
versionCommand: deno --version

> releasePolicyLink: https://github.com/denoland/deno/releases

Reviewer: ```suggestion
releasePolicyLink: https://docs.deno.com/runtime/fundamentals/stability_and_releases/
releaseImage: https://docs.deno.com/runtime/fundamentals/images/deno-lts-support.png
```

changelogTemplate: https://github.com/denoland/deno/releases/tag/v__LATEST__
releaseDateColumn: true

identifiers:
-   purl: pkg:docker/denoland/deno
-   purl: pkg:github/denoland/deno
-   repology: deno

auto:
  methods:
  -   git: https://github.com/denoland/deno.git

releases:
-   releaseCycle: "2"
    releaseDate: 2024-10-09
    eol: false
    latest: "2.1.4"

>     latestReleaseDate: 2024-12-11

Reviewer: Is v1 considered EOL? We should add v1 as well to the table.

Using `eoas` might also be helpful to split between Deno versions receiving active development or only bugfixes.

Author: Added v1 information. There's no guarantees that v1 is supported after it has become non-current (on the v2 release), therefore `eoas` not included. [Source](https://deno.com/blog/v1.46)

---

> [Deno](https://deno.com) is a JavaScript, TypeScript, and WebAssembly runtime with
> secure defaults and a great developer experience. It's built on V8, Rust, and Tokio.

Deno support policy can be found on the

> [Stability and releases documentation page](https://docs.deno.com/runtime/fundamentals/stability_and_releases/).

Reviewer: There needs to be a clear summary of the upstream policy here, instead of just a link. Things to include:

- Release Cadence
- Support policy
- Stability Guarantee
- Release Channels, including the exact guarantees made by LTS releases.
- Versioning Policy(?). Does deno use semver?
- Standard Library stability.

https://endoflife.date/nodejs might be a good reference.

Author: - [x]     Release Cadence
- [x]     Support policy
- [x]     Stability Guarantee
- [x]     Release Channels, including the exact guarantees made by LTS releases.
- [x]     Versioning Policy(?). Does deno use semver?
- [x]     Standard Library stability.

eaaa5d0

---
title: Deno
category: framework

> tags: javascript-runtime

Reviewer 2: ```suggestion
tags: javascript-runtime
iconSlug: deno
```

We can add the icon : https://simpleicons.org/?q=Deno.

permalink: /deno
versionCommand: deno --version
releasePolicyLink: https://docs.deno.com/runtime/fundamentals/stability_and_releases/
releaseImage: https://docs.deno.com/runtime/fundamentals/images/deno-lts-support.png
changelogTemplate: https://github.com/denoland/deno/releases/tag/v__LATEST__
releaseDateColumn: true

identifiers:
-   purl: pkg:docker/denoland/deno
-   purl: pkg:github/denoland/deno
-   repology: deno

auto:
  methods:
  -   git: https://github.com/denoland/deno.git

> releases:
> -   releaseCycle: "2.1"
>     releaseDate: 2024-11-21
>     eoas: 2024-12-21
>     eol: 2025-04-30
>     latest: "2.1.4"
>     latestReleaseDate: 2024-12-11
> -   releaseCycle: "2.0"
>     releaseDate: 2024-10-09
>     eol: 2024-11-21
>     latest: "2.0.6"
>     latestReleaseDate: 2024-11-10
> -   releaseCycle: "1"
>     releaseDate: 2020-04-13
>     eol: 2024-10-09
>     latest: "1.46.3"
>     latestReleaseDate: 2024-09-04

Reviewer 2: ```suggestion
# non-LTS : eol(x) = releaseDate(x+1)
# LTS : eol(x) = releaseDate(x+1) + 5 months
releases:
-   releaseCycle: "2.1"
    lts: true
    releaseDate: 2024-11-21
    eoas: 2024-12-21
    eol: 2025-04-30
    latest: "2.1.4"
    latestReleaseDate: 2024-12-11

-   releaseCycle: "2.0"
    releaseDate: 2024-10-09
    eol: 2024-11-21
    latest: "2.0.6"
    latestReleaseDate: 2024-11-10

-   releaseCycle: "1"
    releaseDate: 2020-04-13
    eol: 2024-10-09
    latest: "1.46.3"
    latestReleaseDate: 2024-09-04
```

- add notes for maintainers,
- flag 2.1 as LTS.

---

> [Deno](https://deno.com) is a JavaScript, TypeScript, and WebAssembly runtime with
> secure defaults and a great developer experience. It's built on V8, Rust, and Tokio.

> Deno uses Semver versioning, offering 4 release channels
> 
> - `stable` (aka `current`): This is the default distribution channel that is recommended for most users.
>   Stable releases are shipped on monthly schedule, each new stable release is a new minor version.
> - `lts`: Long Term Support releases are versions which are maintained for 6 months
>   (stable release monthly period plus 5 months of security and bug fixes).
>   The next LTS release is the one released one month before the previous LTS expiry.
>   LTS releases are recommended users who prefer not to upgrade so often.
> - `rc`: a release candidate for the upcoming semver minor release.
> - `canary`: an unstable release that changes multiple timer per day, allowing users to try out latest
>   bug fixes and new features that might end up in the stable channel.
> 
> Deno maintainers are committed to maintaining a stable standard library API (`Deno` namespace), from
> version `1.0.0` onwards.
> 
> Deno support policy can be found on the
> [Stability and releases documentation page](https://docs.deno.com/runtime/fundamentals/stability_and_releases/).

Reviewer 2: ```suggestion
Deno follows [SemVer](https://semver.org/). New minor releases are made monthly and are supported with bug and security fixes until the next minor release. Every six months a minor version is promoted to LTS and is supported with critical bug and security fixes for an additional 5 months.

Deno maintainers are committed to maintaining a stable standard library API (`Deno` namespace) from
version `1.0.0` onward.
```

What about this ?

The change in this description are:

- A few simplifications.
- No mention to RC and unstable release channels (those are never documented on endoflife.date),
- Drop the mention to the support policy: it is automatically added once the `releasePolicyLink` is set.
